build.gradle

```diff
@@ -177,6 +195,8 @@ dependencies {
     testImplementation group: 'org.objenesis', name: 'objenesis', version: '3.2'
     testImplementation group: 'net.bytebuddy', name: 'byte-buddy-agent', version: '1.14.7'
     testFixturesImplementation "org.opensearch:common-utils:${version}"
+
+    zipArchive group: 'org.opensearch.plugin', name:'opensearch-security', version: "${opensearch_build}"
```

Comment:
Will this add a dependency on security plugin? Like user should install security plugin before installing knn?

Reply:
This will add a dependency for when the user wants to spin up a gradlew cluster using security. 

For instance, if `gradlew run` is executed, it will not require this dependency. Only if `./gradlew run -Dsecurity.enabled=1` is run will it pull it.

I verified this by changing the version to 3.1 (which doesnt exist). When running:
```
./gradlew run -Dsecurity.enabled=1
Picked up JAVA_TOOL_OPTIONS: -Dlog4j2.formatMsgNoLookups=true

FAILURE: Build failed with an exception.

* Where:
Build file '/Users/jmazane/workspace/Opensearch/DockerRunner/k-NN-1/build.gradle' line: 287

* What went wrong:
A problem occurred evaluating root project 'opensearch-knn'.
> Could not resolve all files for configuration ':zipArchive'.
   > Could not find org.opensearch.plugin:opensearch-security:3.1.
     Searched in the following locations:
       - https://repo.maven.apache.org/maven2/org/opensearch/plugin/opensearch-security/3.1/opensearch-security-3.1.pom
       - file:/Users/jmazane/.m2/repository/org/opensearch/plugin/opensearch-security/3.1/opensearch-security-3.1.pom
       - https://aws.oss.sonatype.org/content/repositories/snapshots/org/opensearch/plugin/opensearch-security/3.1/opensearch-security-3.1.pom
       - https://plugins.gradle.org/m2/org/opensearch/plugin/opensearch-security/3.1/opensearch-security-3.1.pom
     Required by:
         project :
```

but `./gradlew run` has no issue.

We can further separate the tasks if we want to provide more separation, but left it like this for development purposes.